string[] material_names
float64[] noise
---
bool[] ok
